CONTRIBUTING.md

```diff
@@ -1,8 +1,123 @@
+```
+
+> Here's a brief introduction to origin and upstream. When we use "git clone", we create an "origin" remote by default, which points to the repository cloned from. In the above example, "origin" is your own forked repo. As for "upstream", we add it ourselves to point to the target repository, which is PyPose's official repo `pypose/pypose`. Of course, you could name it to any arbitrary name other than `upstream`. Usually, contributors will push the code to "origin". If the pushed code conflicts with the latest code in official("upstream"), the contributor should pull the latest code from upstream to resolve the conflicts, and then push to "origin" again. The posted Pull Request will be updated automatically.
+
+### 2. Create a development branch
+After cloning a local copy of the repo, we should create a branch based on the master branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`
+
+```shell
+git checkout -b {your_github_id}/{pr_name}
+```
+
+In subsequent development, if the master branch of the local repository is behind the master branch of "upstream", we need to pull the upstream for synchronization, and then execute the above command:
+
+```shell
+git pull upstream master
+```
+
+### 3. Commit the code and pass the unit test
+- The committed code should pass through the unit test
+
+  ```shell
+  # Pass all unit tests
+  pytest
+  ```
+
+  If the unit test fails for lack of dependencies, you can install the dependencies referring to the [guidance](#unit-test)
+
+- If the documents are modified/added, we should check the rendering result referring to [guidance](#Contributing to Documentation)
+
+### 4. Push the code to remote
+
+We could push the local commits to remote after passing through the check of unit test. You can associate the local branch with your remote branch by adding `-u` option.
+
+```shell
+git push -u origin {branch_name}
+```
+
+This will allow you to use the `git push` command to push code directly next time, without having to specify a branch or the remote repository.
+
+### 5. Create a Pull Request
+
+(1) Create a pull request in GitHub's Pull request interface
+
+(2) Modify the PR description accordingly so that other developers can better understand your changes
+
+**note**
+
+(a) The Pull Request description should contain the reason for the change, the content of the change, and the impact of the change, and be associated with the relevant Issue (see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+
+(b) Check whether the Pull Request pass through the CI
+
+TBD
+
+CI will run unit test for the posted Pull Request on different platforms (Linux, Window, Mac), based on different versions of Python, PyTorch, CUDA to make sure the code is correct. We can see the specific test information by clicking `Details` in the above image so that we can modify the code.
```

Comment:
remove this if we don't have currently.

or use 
<img width="909" alt="image" src="https://user-images.githubusercontent.com/8695500/222796987-3f612b21-21c9-4bb4-b8ff-4b1ddccf115d.png">
by  using this code
```
<img width="909" alt="image" src="https://user-images.githubusercontent.com/8695500/222796987-3f612b21-21c9-4bb4-b8ff-4b1ddccf115d.png">
```